server/index.js

```diff
@@ -12,64 +12,77 @@ const app = express()
-    'lott': { latitude: 30.271041206071896, longitude: -97.7297671152558 },
-    'metz': { latitude: 30.252258120681006, longitude: -97.71810034083994 },
-    'mueller-branch-park': { latitude: 30.300190, longitude: -97.702812 },
-    'ricky-guerrero': { latitude: 30.247392178325605, longitude: -97.76433234845857 },
-    'rosewood': { latitude: 30.27109617907063, longitude: -97.71390894361079 },
-  }[key]
+const getParkLocation = R.memoizeWith(
+  R.identity,
+  async (parkName) => await knex.select('latitude', 'longitude').from('parks').where({ name: parkName }).first()
 )
 
+class DBStatusRepository {
 
-class StatusRepository {
   constructor() {
-    this.votes = db.addCollection('votes');
+    this.getParkId = R.memoizeWith(R.identity, async parkName => (await knex.select('id').from('parks').where({ name: parkName }).first()).id)
   }
 
-  addOnVote(weight, parkKey) {
+  async addOnVote(weight, parkName) {
+    const park_id = await this.getParkId(parkName)
     for (var i = 0; i < weight; i++)
-      this.votes.insertOne({ status: true, timestamp: new Date(), parkKey })
+      await knex('votes').insert({ status: true, park_id })
   }
 
-  addOffVote(weight, parkKey) {
+  async addOffVote(weight, parkName) {
+    const park_id = await this.getParkId(parkName)
     for (var i = 0; i < weight; i++)
-      this.votes.insertOne({ status: false, timestamp: new Date(), parkKey })
+      await knex('votes').insert({ status: false, park_id })
   }
 
-  getLastVoteTime(parkKey) {
-    const voteDates = this.votes
-      .find({ timestamp: { $gte: twentyFourHoursAgo() }, parkKey })
-      .map(v => v.timestamp)
+  async getLastVoteTime(parkName) {
 
-    if (voteDates.length > 0)
-      return timeAgo.format(voteDates.reduce(function (a, b) { return a > b ? a : b; }))
+    const park_id = (await knex.select('id').from('parks').where({ name: parkName }).first()).id
+    const lastVoteRecord = await knex
+      .select('created_at')
+      .where({ park_id })
+      .where('created_at', '>=', new Date(twentyFourHoursAgo()))
+      .from('votes')
+      .orderBy('created_at', 'desc')
+      .first()
 
-    return 'N/A'
+    return lastVoteRecord ? timeAgo.format(lastVoteRecord.created_at) : 'N/A'
   }
 
-  getStatus(parkKey) {
-    const workingVotes = this.votes.count({ status: true, timestamp: { $gte: twentyFourHoursAgo() }, parkKey })
-    const notWorkingVotes = this.votes.count({ status: false, timestamp: { $gte: twentyFourHoursAgo() }, parkKey })
-
-    if (workingVotes === 0 && notWorkingVotes === 0) {
-      const orderedVotes = this.votes.chain().find({ parkKey }).simplesort('timestamp', { desc: true }).data()
-      if (orderedVotes.length === 0 || orderedVotes[0].status === false) {
-        return { status: "unknown", workingVotes, notWorkingVotes }
+  async getStatus(parkName) {
+    const park_id = await this.getParkId(parkName)
+    const lastVoteRecords = await knex
```

Comment:
there are probably better ways to do these queries, but this works for now. thoughts?

Reply:
I'm actually new to the knex query builder. This looks good for now though!